### PR TITLE
Separate subprocess writing/parsing into their own functions

### PR DESCRIPTION
**Pull request by mdboom**
...and document

As suggested by @corona10 https://github.com/psf/pyperf/pull/197#discussion_r1718097565

### CODE REVIEW

File: pyperf/_process_time.py [hunk 1]
```diff
@@ -114,6 +114,18 @@ def load_hooks(metadata):
...
+def write_data(dt, max_rss, metadata, out=sys.stdout):
+    # The data that is communicated back to the main orchestration process.
+    # It is three lines containing:
+    #    - The runtime (in seconds)
+    #    - max_rss (or -1, if not able to compute)
+    #    - The metadata to add to the benchmark entry, as a JSON dictionary
+    # Write timing in seconds into stdout
+    print(dt, file=out)
+    print(max_rss or -1, file=out)
+    print(json.dumps(metadata), file=out)
```
Comment (vstinner) [inline pyperf/_process_time.py:126]: Can you call .dump() on sys.stdout?

Comment (mdboom) [reply pyperf/_process_time.py]: Sure, that makes sense.

File: pyperf/_command.py [hunk 2]
```diff
@@ -8,6 +8,26 @@
...
+
+    rss = None
+    metadata = {}
+    try:
+        lines = output.splitlines()
+        timing = float(lines[0])
+        rss = int(lines[1])
+        metadata = json.loads(lines[2])
+    except ValueError:
+        raise ValueError("failed to parse script output: %r" % output)
```
Comment (vstinner) [inline pyperf/_command.py:26]:
```suggestion
        raise ValueError("failed to parse worker output: %r" % output)
```

**Review by vstinner — APPROVED** [review]
LGTM.

**Review by corona10 — APPROVED** [review]
LGTM, Thank you for adopting my suggestions!

**Pull request merged**